src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -89,7 +91,9 @@ public String toString() {
     public static SamInputResource of(final File file) { return new SamInputResource(new FileInputResource(file)); }
 
     /** Creates a {@link SamInputResource} reading from the provided resource, with no index. */
-    public static SamInputResource of(final Path path) { return new SamInputResource(new PathInputResource(path)); }
```

Comment:
Kept overload, added comment here. Which other ctors would you like to add a comment to? I think this one's the only public one so it would be the right place for the comment.